pkg/envoy/rds/route/route_config.go

```diff
@@ -286,6 +287,40 @@ func buildWeightedCluster(weightedClusters mapset.Set) *xds_route.WeightedCluste
 	sort.Stable(clusterWeightByName(wc.Clusters))
 	return &wc
 }
+func buildRetryPolicy(retry *v1alpha1.RetryPolicySpec) *xds_route.RetryPolicy {
+	rp := &xds_route.RetryPolicy{}
+	if retry == nil {
+		return rp
+	}
+	if retry.RetryOn != "" {
+		rp.RetryOn = retry.RetryOn
+	} else {
+		log.Error().Msgf("Retry policy must specify which policies to retry on")
+		return rp
+	}
+	if retry.NumRetries != 0 {
+		rp.NumRetries = &wrapperspb.UInt32Value{
+			Value: uint32(retry.NumRetries),
+		}
+	}
+	if retry.PerTryTimeout != 0 {
+		rp.PerTryTimeout = convertIntToDuration(retry.PerTryTimeout)
+	}
+	if retry.RetryBackoffInterval != 0 {
+		rp.RetryBackOff = &xds_route.RetryPolicy_RetryBackOff{
+			BaseInterval: convertIntToDuration(retry.RetryBackoffInterval),
+		}
+	}
+	return rp
+}
+
+func convertIntToDuration(n int) *durationpb.Duration {
```

Comment:
What does the int value represent? What is the unit of time here? Kindly add a comment.

Reply:
Added comment, lmk if it needs further clarification 
"// millisecondToNanosecond converts integer(ms) into durationpb.Duration(ms)
// durationpb.New() creates a *durationpb.Duration in Nanoseconds"